geocoding/generate_pincode.py

```diff
@@ -18,91 +18,153 @@
-        api_key (str): API Key for using GMap API
+        csv_sheet (str): CSV sheet that needs to be reverse geocoded for pincodes
+        api_key (str): API Key to access the GMap API
+        output_file (str, optional): Name of the new CSV to be stored as. Defaults to 'AssamNewAddressPincode'.
     """
     gmaps = googlemaps.Client(key=api_key)
     csv_sheet = pathlib.Path.cwd() / csv_sheet
     df = pd.read_csv(csv_sheet)
     lats, lngs = df["Latitude"].tolist(), df["Longitude"].tolist()
     postal_codes, pincode_verify = [], []
     is_assam = []
-    count = 0
-    nancount = 0
-    success = 0
-    index = 0
-    for lat, lng in tqdm(zip(lats, lngs)):
-        index += 1
+    nocode_count, nancount, success = 0, 0, 0
+
+    def verify_ifnan(lat, lng):
+        """Simply verifies if the given lat, lng is nan or not.
+        If nan. Appends the neccessary content to three lists
+        
+        Args:
+            lat (float): Latitude
+            lng (float): Longitude
+        
+        Returns:
+            Boolean : True if the sent lat, lng were nan
+        """
         if math.isnan(lat) or math.isnan(lng):
             pincode_verify.append(True)
             postal_codes.append(None)
             is_assam.append(None)
             logging.info("Could not find pincode for NaN values")
             nancount += 1
+            return True
+        return False
+
+    def get_required_code(codes):
+        """Takes the List of all the possible pincodes and selects the pincode for Assam
+
+        TODO: Maintain a mapping of state to pincode and take the state as input.
+        
+        Args:
+            codes (List): Collection of all pincodes reccieved from the API
+        """
+        if len(codes) == 0:
+            logging.warning(f"API returned no postal code value - {(lat,lng)}")
+            postal_codes.append(None)
+            is_assam.append(None)
+            nocode_count += 1
+            return
+        for code in codes:
+            if str(code)[0] == "7":
+                postal_codes.append(code)
+                is_assam.append(True)
+                return
+        postal_codes.append(codes[0])
+        is_assam.append(False)
+        return
+
+    test = [True, True, False, False, True]
+    index = 0
+    for lat, lng in tqdm(zip(lats, lngs)):
+        if verify_ifnan(lat, lng):
             continue
 
-        try:
-            reverse_geocode_result = gmaps.reverse_geocode((lat, lng))
-            pincode_verify.append(True)
-        except:
-            logging.warning(f"Failure in reaching the API - {(lat,lng)}")
+        reverse_geocode_result, successful_api_hit = send_reverese_geocoding_request(
+            lat, lng, gmaps
+        )
+
+        successful_api_hit = test[index]
+        index += 1
+
+        if not successful_api_hit:
             pincode_verify.append(False)
             postal_codes.append(None)
             is_assam.append(None)
+            logging.info("Here??")
```

Comment:
Do you think we can probably have something more descriptive -- other than `Here??` ?

Reply:
My bad. Removed the logging.